sklearn/decomposition/_nmf.py

```diff
@@ -712,14 +717,26 @@ def _multiplicative_update_h(X, W, H, beta_loss, l1_reg_H, l2_reg_H, gamma):
         denominator = denominator + l2_reg_H * H
     denominator[denominator == 0] = EPSILON
 
-    numerator /= denominator
-    delta_H = numerator
-
-    # gamma is in ]0, 1]
-    if gamma != 1:
-        delta_H **= gamma
+    if A is not None and B is not None:
+        if gamma != 1:
+            H **= 1 / gamma
+        numerator *= H
+        A *= rho
+        B *= rho
+        A += numerator
+        B += denominator
+        H = A / B
+
+        if gamma != 1:
+            H **= gamma
+    else:
+        delta_H = numerator
+        delta_H /= denominator
+        if gamma != 1:
+            delta_H **= gamma
+        H *= delta_H
```

Comment:
Well the origin is in the referenced articles in the docstring. I think it would be redundant to repeat the refs here and there.
For the non-online updates, "Algorithms for Nonnegative Matrix Factorization with the β-Divergence is what you're looking for". and for the online version it's "Online algorithms for nonnegative matrix factorization with the Itakura-Saito divergence".